### PR TITLE
Fix for running in pythonw

### PR DESCRIPTION
**Pull request by timmo001**
Running sanic server on windows and on `pythonw.exe` causes the app to silently crash due to `sys.stdout` being None.

NVIDIA_Share_2022-05-04_12-36-03
 

This adds a check for stdout. The rest of the server then continues to function normally.

### CODE REVIEW

File: sanic/mixins/runner.py
Comment (ahopkins) [inline sanic/mixins/runner.py:427]: For some reason GH is not giving me the option to create a suggestion here... :thinking: 

Nevetheless, can you change this whole block like this:

```python
        if not self.state.is_debug:
            message = (
                "Sanic is running in PRODUCTION mode. Consider using "
                "'--debug' or '--dev' while actively developing your "
                "application."
            )
            if sys.stdout and sys.stdout.isatty():
                message = "".join([Colors.YELLOW, message, Colors.END])
            error_logger.warning(message)
```

**Review by ahopkins — CHANGES_REQUESTED** [review]

File: sanic/mixins/runner.py
Comment (ahopkins) [reply sanic/mixins/runner.py:427]: At some point we can abstract some of the color stuff, but for now this at least keeps the message still populating.

**Comment by ahopkins** [comment]
@timmo001 I am sorry, in my head I was looking at the original logic thinking that it was wrong. That is why I suggested the change. But, it was correct. I reverted the PR back to your original suggestion. Let's pretend I never suggested that to begin with :laughing: 

Thanks for catching this for us.

**Review by ahopkins — APPROVED** [review]

**Review by ahopkins — CHANGES_REQUESTED** [review]
Here's a suggestion that _ACTUALLY_ makes sense...

There are a few more places that follow a similar pattern. Perhaps you can add the same change:

```
sanic/mixins/runner.py:427:        if sys.stdout and sys.stdout.isatty() and not self.state.is_debug:
sanic/mixins/runner.py:618:                if sys.stdout.isatty():
sanic/mixins/runner.py:659:                        if sys.stdout.isatty():
sanic/log.py:86:    if sys.stdout.isatty():
sanic/application/motd.py:39:        motd_class = MOTDTTY if sys.stdout.isatty() else MOTDBasic
sanic/application/logo.py:47:        if sys.stdout.isatty()
```

**Review by ahopkins — DISMISSED** [review]

**Comment by timmo001** [comment]
Update on this? I see its been approved but the checks are failing.

I've rebased, but this requires an approver to run

**Comment by timmo001** [comment]
I've rebased, but this requires an approver to run

**Comment by timmo001** [comment]
Fixed lint, needs approval

**Review by ahopkins — APPROVED** [review]

**Review by ahopkins — APPROVED** [review]

**Review by ahopkins — DISMISSED** [review]

**Comment by timmo001** [comment]
Schedule for the next release?

**Comment by ahopkins** [comment]
Should be this week.

**Pull request merged**
